core/src/test/scala/org/apache/spark/rpc/RpcEnvSuite.scala

```
@@ -162,9 +163,15 @@ abstract class RpcEnvSuite extends SparkFunSuite with BeforeAndAfterAll {
     val rpcEndpointRef = anotherEnv.setupEndpointRef("local", env.address, "ask-timeout")
     try {
       val e = intercept[Exception] {
-        rpcEndpointRef.askWithRetry[String]("hello", 1 millis)
+        rpcEndpointRef.askWithRetry[String]("hello", new RpcTimeout(1 millis, shortProp))
       }
       assert(e.isInstanceOf[TimeoutException] || e.getCause.isInstanceOf[TimeoutException])
+      e match {
+        case te: TimeoutException =>
+          assert(te.getMessage().contains(shortProp))
+        case e: Exception =>
+          assert(e.getCause().getMessage().contains(shortProp))
```

The call to `askWithRetry` will catch all exceptions and then throw a `SparkException` with the `TimeoutException` as the cause.  So its the `TimeoutException` case that will never happen, but I left that in there to keep it as before.